### ISSUE TITLE
line break toggle covers code block content

### ISSUE THREAD

**Issue by n1ru4l** [the-guild-org/docs#1425]
line break toggle covers code block content

![image](https://github.com/the-guild-org/docs/assets/14338007/bf0b2ac2-4a3f-47fc-9058-6ced42a58467)

**Comment by dimaMachina** [comment]
I remember you already submitted similar issue in v2.
In v3 I
- moved copy code button in code block title
- removed blured background
- added transparent bg instead similar to docusaurus

What’s your proposition to fix it for now?

**Comment by n1ru4l** [comment]
Possible options are
- hide completely
- collapse both buttons into a context menu

**Comment by dimaMachina** [comment]
> hide completely

disagree, useful feature, especially on mobile

> collapse both buttons into an expandable context/dropdown menu

I think it's too overcomplicated for only 2 actions

---


As an another idea we can do approach similar as GitHub do but we'll lose some `44px` for code-block content

<img width="482" alt="image" src="https://github.com/the-guild-org/docs/assets/7361780/81076feb-edb1-4c26-aa63-9de3aa184753">

**Comment by n1ru4l** [comment]
I rarely use the copy and line-break toggle on my phone, so I disagree that a context menu would be too complicated. 

How often do people copy a code snippet on mobile, for me it makes most sense on a desktop where you want to run the example? It might be an acceptable trade-off to hide that button and only display the line break toggle.

The code area is already quite small due to all the vertical spacings, so the latter seems suboptimal.

Reducing the size of the buttons in the header and placing them next to each other could also be a possibility.

**Comment by dimaMachina** [comment]
> How often do people copy a code snippet on mobile, for me it makes most sense on a desktop where you want to run the example? It might be an acceptable trade-off to hide that button and only display the line break toggle.

this is true, we already show `Word wrap` button only on mobile, so we can hide `Copy button` from mobile and put `Word wrap` there

> The code area is already quite small due to all the vertical spacings, so the latter seems suboptimal.

true too

> Reducing the size of the buttons in the header and placing them next to each other could also be a possibility.

We can try to see how it looks like, but I would stay with putting word wrap button on copy button place

Accepted to be fixed in Nextra directly!

**Comment by dimaMachina** [comment]
@n1ru4l I've just thought that github approach will be perfect since if code block doesn't contain filename, copy code button will be in code block and not in its filename, so we can include copy-code for mobile as well (since github has it on mobile too)

**Comment by Urigo** [comment]
The new site's code blocks have a single floating copy button that only shows on hover or focus, and no line-break toggle, so nothing covers the code any more.
